### DISCUSSION TITLE
Требуются пояснения по синтаксису.

### DISCUSSION THREAD

**Discussion by ebkjiud** [hufrea/byedpi#76] [General]
Не очень понятно из README, как правильно указывать значения для опций. Правильны ли подобные конструкции? (Если не сложно, уточните, где ошибки.)

1.  "-t 8"
2.  "-n www.iana.org"
3.  "-f -1"
4. "-o 2+s"
5.  "-d 3+s"
6.  "-s 1+s"
7.  "-r 3+s"
8.  "-M h,d"

• Когда необходимо указывать параметры через "--" и значение через "="?

**Дополнительно, если не сложно, прокомментируйте следующие вопросы:**
• Для каких параметров применимы смещения +s и +h?
• Какова логика -A? Имеет ли смысл указывать много групп (до 20, например) без фильтрации по портам и хостам?
• Какие эффекты имеют -u и -a?
• Не совсем понятно, какие параметры несовместимы в одной группе. 

Пожалуйста, дайте пояснения по этим вопросам.

**Comment by hufrea** [comment]
> Правильны ли подобные конструкции

Да
> Когда необходимо параметры указывать через "--" и указывать значение через "="?

Если значение опционально, в выводе `--help` аргументы к таким опциям обрамлены квдратными скобками (сейчас это `--auto` и `--ip-opt`).
>  Для каких параметров применимо смещения +s и +h?

Для тех, где аргументом указан `<n[+s]>`
>  Еще интересует логика -A, имеет ли эффективность указать много групп до 20 к примеру (без фильтрации по портам и хостам)?

Без фильтров и триггеров будет работать только первая группа. Если же испольпользуется какой-либо триггер (torst, redirect), то будет создаваться новое поключение с каждым переходом на новую группу, пока действует триггер.

>  Какие эффекты имеют -u и -a?

Чтобы не скакать по группам каждый раз, для конкретного IP кеширует индекс группы. -u определяет сколько времени будет храниться значение в кеше.
-a указывает сколько фейковых UDP пакетов отправить перед настоящим.

> Не совсем понятно, какие параметры несовместимы в одной группе.

https://github.com/hufrea/byedpi/blob/084921e523fdb1872d6f09eb591436371c922c29/readme.txt#L217-L218
Эти параметры не конфликтуют между собой.

**Comment by eddiezato** [comment]
@hufrea пока подбирал параметры, столкнулся с тем, что
- так работает: `--auto=none` и `--auto=n`
- так **не** работает: `-A none` и `-A n`

И еще вопрос. Необходимо для одного домена одну стратегию, для другого иную, для всех остальных не запутывать, такой набор параметров верный?
```
-H ":rutracker.org" -f -1 -t 7 --auto=n -H ":googlevideo.com" -d 1 -t 7 --auto=n
```
Смысл в том, чтобы для остальных неуказанных доменов не применялось никакой стратегии "по умолчанию".

**Reply by ebkjiud** [reply]
Может имеет смысл тогда сначало по bypass пускать, то есть --auto=n в начале запуска, я такую логику использую, как работает не уверн.

**Reply by eddiezato** [reply]
Я ориентировался на пример из readme:
> --hosts list.txt --disorder 3 --auto=none  
Применять запутывание только для доменов из list.txt

**Reply by ebkjiud** [reply]
-u 120 -a 2 -T 0.25 --auto=t,r,c,s,a,n -f -1 -r 2+s -n www.iana.org -t 8 --auto=t,r,c,s,a,n -d 2+s
 
У меня так сейчас запускает и еще несколько групп дальше...
Вообще смотря чем заруливаешь в этот сокс трафик, у меня перед этим еще маркировка, можно на этом этапе часть фильтрации сделать.

**Reply by hufrea** [reply]
> такой набор параметров верный?

Да.
Только во второй группе `-t` лишний, там нет фейка.

**Reply by eddiezato** [reply]
Я ориентировался на стратегии, которые blockcheck из zapret'а мне подсказывал, там одна из стратегий *disorder + ttl 7*.

В итоге упростил до `-H ":googlevideo.com" -d 1 -S --auto=n -H ":example.com" -f -1 --auto=n`. В таком варианте работает пока лучше всего, и ютуб, и другой домен в браузере, и всему остальному не мешает.
